File: kirun-java/src/main/java/com/fincity/nocode/kirun/engine/json/schema/convertor/NullConvertor.java
package com.fincity.nocode.kirun.engine.json.schema.convertor;

import static com.fincity.nocode.kirun.engine.util.json.ConvertorUtil.handleUnConvertibleValue;

import java.util.List;

import com.fincity.nocode.kirun.engine.json.schema.Schema;
import com.fincity.nocode.kirun.engine.json.schema.convertor.enums.ConversionMode;
import com.google.gson.JsonElement;
import com.google.gson.JsonNull;
import com.google.gson.JsonPrimitive;

public class NullConvertor {

	public static JsonElement convert(List<Schema> parents, Schema schema, ConversionMode mode, JsonElement element) {

		if (element == null || element.isJsonNull())
			return JsonNull.INSTANCE;

		if (element.isJsonPrimitive()) {
			JsonPrimitive primitive = element.getAsJsonPrimitive();
			if (primitive.isString() && primitive.getAsString().equalsIgnoreCase("null"))
				return JsonNull.INSTANCE;
		}

		return handleUnConvertibleValue(parents, mode, element, JsonNull.INSTANCE, "Unable to convert to null");
	}

	private NullConvertor() {
	}
}

File: kirun-js/src/engine/json/schema/convertor/NullConvertor.ts
import { Schema } from '../Schema';
import { ConversionMode } from './enums/ConversionMode';
import { isNullValue } from '../../../util/NullCheck';
import { ConvertorUtil } from '../../../util/json/ConvertorUtil';

export class NullConvertor {
    public static convert(
        parents: Schema[],
        schema: Schema,
        mode: ConversionMode,
        element: any,
    ): null {
        if (isNullValue(element)) {
            return element;
        }

        if ('string' === typeof element && element.toLowerCase() === 'null') {
            return null;
        }

        return ConvertorUtil.handleUnConvertibleValueWithDefault(
            parents,
            mode,
            element,
            null,
            'Unable to convert to null',
        );
    }

    private constructor() {}
}
